doc: Add an Gazell section in the memory optimization guide. (#8543)
nrfconnect/sdk-nrf: doc/nrf/app_memory.rst

For reference, you can find minimal footprint configurations of the :ref:`peripheral_lbs` sample in :file:`nrf/samples/bluetooth/peripheral_lbs/minimal.conf` and the :ref:`peripheral_uart` sample in :file:`nrf/samples/bluetooth/peripheral_uart/minimal.conf`.

.. _app_memory_gazell:

Gazell
******

To optimize the Gazell memory footprint, follow the `General recommendations`_.

> Specifically, study the ISR stack size.
> If your application is a Pairing Device, you may study the system workqueue stack size.

Reviewer: ```suggestion
Specifically, study the Interrupt Service Routine (ISR) stack size.
If your application is in a pairing device, pay attention to the system workqueue stack size.
```

> Do not enable the features that your application does not use. The configuration options default to disable the optional features.

Reviewer: ```suggestion
Do not enable features that your application does not use.
The configuration options default to disabling optional features, such as:
```

* Pairing
* Pairing encryption
* Pairing settings persistent storage

> You may reduce the logging level by the setting configuration option :kconfig:option:`CONFIG_GAZELL_LOG_LEVEL_CHOICE`.

Reviewer: ```suggestion
To reduce the logging level, set the :kconfig:option:`CONFIG_GAZELL_LOG_LEVEL_CHOICE` Kconfig option.
```
